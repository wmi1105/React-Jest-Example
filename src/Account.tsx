export function Account(props: { name: string; mbti: string }) {
  return (
    <div>
      <h2>{props.name}</h2>
      <p>{props.mbti}</p>
    </div>
  );
}
