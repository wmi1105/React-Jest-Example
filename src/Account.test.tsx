/* eslint-disable testing-library/prefer-screen-queries */
import { Account } from "./Account";
import { render } from "@testing-library/react";

it("matched snapshot", () => {
  const utils = render(<Account name="호박너구리" mbti="ESFJ" />);
  expect(utils.container).toMatchSnapshot(); //컴포넌트를 수정했을 때 원하는 방식으로 렌더되는지 비교
});

it("shows the props correctly", () => {
  const utils = render(<Account name="호박너구리" mbti="ESFJ" />);
  utils.getByText("호박너구리"); //특정 문구가 있는지 확인
  utils.getByText("ABCD");
});
