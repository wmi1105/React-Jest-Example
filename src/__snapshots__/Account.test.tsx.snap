// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<div>
  <div>
    <h2>
      호박너구리
    </h2>
    <p>
      ESFJ
    </p>
  </div>
</div>
`;
